pkg/kubelet/kuberuntime/kuberuntime_container_windows.go

```diff
@@ -79,22 +84,32 @@ func (m *kubeGenericRuntimeManager) generateWindowsContainerConfig(container *v1
 		//   https://github.com/kubernetes/kubernetes/blob/56d1c3b96d0a544130a82caad33dd57629b8a7f8/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto#L681-L682
 		//   https://github.com/opencontainers/runtime-spec/blob/ad53dcdc39f1f7f7472b10aa0a45648fe4865496/config-windows.md#cpu
 		//   If both CpuWeight and CpuMaximum are set - ContainerD catches this invalid case and returns an error instead.
-		wc.Resources.CpuMaximum = calculateCPUMaximum(cpuLimit, int64(winstats.ProcessorCount()))
+		resources.CpuMaximum = calculateCPUMaximum(cpuLimit, int64(winstats.ProcessorCount()))
 	}
 
 	// The processor resource controls are mutually exclusive on
 	// Windows Server Containers, the order of precedence is
 	// CPUCount first, then CPUMaximum.
-	if wc.Resources.CpuCount > 0 {
-		if wc.Resources.CpuMaximum > 0 {
-			wc.Resources.CpuMaximum = 0
+	if resources.CpuCount > 0 {
+		if resources.CpuMaximum > 0 {
```

Comment:
Do we ever enter this block?

Reply:
The part with `if wc.Resources.CpuMaximum > 0' was introduced here [#97141](https://github.com/kubernetes/kubernetes/pull/97141). Before that, the `!isolatedByHyperv` was created in [#85856](https://github.com/kubernetes/kubernetes/pull/85856) based on this issue [#84804](https://github.com/kubernetes/kubernetes/issues/84804)

Reply:
We should untangle this as a separate change IMO

Reply:
SG